go/frontendbridge/main.go

```diff
@@ -69,15 +71,15 @@ func (w *FrontendBridge) addConnection(auth structs.MessageAuth, socket *websock
 }
 
 func (w *FrontendBridge) getConnection(auth structs.MessageAuth) *Connection {
-	w.connectionsMutex.Lock()
+	w.connectionsMutex.RLock()
 	var result *Connection
 	for _, connection := range w.connections {
 		if reflect.DeepEqual(auth, connection.auth) {
 			result = connection
 			break
 		}
 	}
-	w.connectionsMutex.Unlock()
+	w.connectionsMutex.RUnlock()
```

Comment:
You are doing the entire loop inside the lock. I'm not 100% sure what the best way to use locks is, but I think I remember that you should keep the code path inside the lock as short as possible, since locks are global bottlenecks.